rfd/0123-tls-routing-behind-layer7-lb.md

```diff
@@ -228,6 +228,30 @@ See [RFD 0121 - Kubernetes MFA
 sessions](https://github.com/gravitational/teleport/blob/master/rfd/0121-kube-mfa-sessions.md)
 for more details.
 
+### Client source IPs
+
+Correct client source IPs must be reported for IP pinning and audit logs.
+
+Unlike layer 4 load balancers, layer 7 load balancers usually CANNOT transparently forward original clients' IPs or add
+standard [PROXY](https://www.haproxy.org/download/1.8/doc/proxy-protocol.txt) protocol headers to layer 7 payload. The
+standard way for layer 7 load balancers to report correct client source IPs is through the
+["X-Forwarded-For"](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-For) (XFF) header.
+
+The Proxy service will examine the values of the XFF headers when `proxy_service.use_xff_header` is enabled (more
+details in security section).
+
+ALL web APIs will take client source IPs from the XFF headers. This includes login APIs, Web UI APIs (including wss
+sessions) and the ALPN connection upgrade API, to make sure a correct IP is always logged.
+
+Teleport should support XFF header with both IPv4 and IPv6 addresses, with or without port numbers. In case no port is
+reported in the XFF header, port can be taken from the observed TCP connection or use an arbitrary number, since the
+port is usually trivial.
+
+In case of multiple IPs are appended in the header, Teleport takes the last IP in the list. If the client request has to
+go through multiple jumps of reverse proxies/load balancers, say request -> jump A -> jump B -> jump C -> Teleport
+Proxy, only the first jump A should be setting the XFF header and the user can configure jump B and jump C to maintain
+the header values without modifying it.
```

Comment:
Yes, only the last value should be used. But in the case of chained ALB/reverse proxies, if the first one (closest to client) is in APPEND mode, later ones (closer to Proxy) in PRESERVE mode, the value should also be correct? assuming  client may not be trusted but everything in this chain can be trusted,

Reply:
How would you distinguish the number of chained ALBs when users are allowed to fake their own XFF headers?

Reply:
Can we just narrow the scope to support one L7 LB and in case of  multiple values `X-Forwarded-For` we will just block the request.

Reply:
Agree. it was my initial idea since it's not possible to correctly validate the other cases.

We must limit the number of ALBs to 1 and use the last appended address as the IP address. 
If multiple XFF values are received, we deny the request.